prow/flagutil/config.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flagutil
+
+import (
+	"flag"
+	"fmt"
+
+	"k8s.io/test-infra/prow/config"
+)
+
+// ConfigOptions holds options for interacting with Configs.
+type ConfigOptions struct {
+	ConfigPath       string
+	JobConfigPath    string
+	PluginConfigPath string
+
+	// ConfigPathNotEmpty is a boolean that enables validation of ConfigPath string.
+	ConfigPathNotEmpty bool
+	// JobConfigPathNotEmpty is a boolean that enables validation of JobConfigPath string.
+	JobConfigPath bool
+	// PluginConfigPathNotEmpty is a boolean that enables validation of PluginConfigPath string.
+	PluginConfigPath bool
```

Comment:
nit: This seems to me like we are shoehorning configs together. Plugin config is handled similarly to core config, but I don't think there is any benefit to combining it into this struct. I think having a `ConfigOptions` and a `PluginConfigOptions` might make more sense given that some things need config, some need plugin config, and some need both.
Similarly we could split `ConfigOptions` into a `CoreConfigOptions` and a `JobConfigOptions` (the latter containing a `CoreConfigOptions` since job config must be loaded with core config) so that binaries could use precisely the options that apply to them instead of a potentially partially initialized options struct.

Reply:
That works for me... take a look again please.